diff --git a/src/pl-comp.c b/src/pl-comp.c
index 1cbc141..4519106 100644
--- a/src/pl-comp.c
+++ b/src/pl-comp.c
@@ -7310,8 +7310,9 @@ clause(term_t head, term_t body, term_t ref, term_t bindings, int flags,
 	{ term_t tmp;
 	  int hflags = 0;
 
-	  if ( rc < 0 && CTX_ARITY < 4 )
-	    return false;			/* erased clause */
+	  if ( rc < 0 && CTX_ARITY < 4 &&
+	       !visibleClause(clause, current_generation(clause->predicate)) )
+	    return false;			/* erased, and not visible here */
 
 	  if ( protected_predicate(clause->predicate) )
 	    return false;
